src/explorer/CosmosDBTreeItem.ts

```diff
@@ -159,10 +171,48 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
         return result;
     }
 
+    private detectPostgresConnections(appSettings: { [propertyName: string]: string }): IDetectedConnection[] {
```

Comment:
This code is pretty similar to `detectDocDBConnetions`.  I wonder if we could do something to share some more code.

I was thinking maybe you could make a helper function that accepts the prefix and suffixes as parameters, and then it can iterate over these arrays.  For instance, with Postgres, you could pass `[_pgHostSuffix,  _pgDbNameSuffix, _pgUserSuffix, _pgPassSuffix, _pgPortSuffix]` and then build all of the appSettings out that you need?

Reply:
I like this idea, given detectPostgresConnections is too long.

Reply:
I tried to simplify the prefix and suffix creation part however it was just creating more redundant code. Even if we can the list of final keys (prefix + suffix) we still need to store them as individual variables in order to access the values from `appSettings` to create the connection string.